src/Main.vala

```diff
@@ -1,28 +1,56 @@
 using Gtk;
+using GtkLayerShell;
+
+// Globals
+bool IS_SESSION_WAYLAND;
+string WM_NAME;
 
 /**
  * Application entry point
  */
 public static int main (string[] args) {
     Gtk.init (ref args);
 
+    // Get session type (wayland or x11) and set the flag
+    string session_type = Environment.get_variable ("XDG_SESSION_TYPE");
+    string gdk_backend = Environment.get_variable ("GDK_BACKEND");
+    IS_SESSION_WAYLAND = session_type == "wayland" && gdk_backend != "x11";
+
+    // Set window manager
+    string sway_sock = Environment.get_variable ("SWAYSOCK");
+    string i3_sock = Environment.get_variable ("I3SOCK");
+
+    if(sway_sock != null) {
```

Comment:
question: why not base socket path on `IS_SESSION_WAYLAND` rather than checking for files?  (single source of truth)

Reply:
Can you clarify what you mean by base socket path?

Reply:
I think it would make more sense to check the `XDG_SEESSION_TYPE` env vars instead of the socket. Using a socket  (`SWAYSOCK`) would cause the wayland specific code of `ilia` to execute only when the window manager is `sway` even though ilia should work with any other `wlroots` based compositor as well.

Reply:
Btw theres a chance I misunderstood the question. I didn't actually get what you meant by "checking for files".